### DISCUSSION TITLE
Is there any way to get useful errors out of the PowerPlatform build tools

### DISCUSSION THREAD

**Discussion by dreashovlain** [microsoft/powerplatform-build-tools#1082] [Q&A]
I'm using ADO and the PowerPlatform build tools to deploy solutions between environments. Howevever the errors are horrible. 99.9% of the time, the only error I can find in my output is:

> 2024-11-21T16:48:40.1445515Z [ 'The reason given was: An unexpected error occurred.' ]
...
2024-11-21T16:48:40.4642910Z ##[error]error: 1
2024-11-21T16:48:40.4655663Z ##[error]failed: Connected as ***
>

Is there a setting or flag I can set to get more detailed errors? the debug option on the pipelines doesn't provide anything better.

**Comment by petrochuk** [comment]
yes, set "enable system diagnostics" when you run your pipeline to get more detailed logging

<img width="338" alt="image" src="https://github.com/user-attachments/assets/14b41c81-ffc4-4b48-bf4d-0df480a14758">

**Reply by dreashovlain** [reply]
I don't get that option on a release pipeline

**Comment by star-crossed** [comment]
From [Use predefined variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemdebug):

```
For more detailed logs to debug pipeline problems, define System.Debug and set it to true.
1. Edit your pipeline.
2. Select Variables.
3. Add a new variable with the name System.Debug and value true.
```
